### PR TITLE
pywin32: Fix clang build

### PR DESCRIPTION
**Pull request by raedrizqie**

### CODE REVIEW

File: mingw-w64-python-pywin32/PKGBUILD [hunk 1]
```diff
@@ -10,29 +10,27 @@ url='https://sourceforge.net/projects/pywin32'
 pkgdesc='Python for Windows Extensions (mingw-w64)'
 license=('PSF')
 arch=('any')
-mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang32' 'clang64' 'clangarm64')
 pkgver=301
 pkgrel=1
```
[the remark this answers is not in the document]
Comment (lazka) [reply mingw-w64-python-pywin32/PKGBUILD:15]: bumped in master

File: mingw-w64-python-pywin32/PKGBUILD [hunk 2]
```diff
@@ -10,29 +10,27 @@ url='https://sourceforge.net/projects/pywin32'
 pkgdesc='Python for Windows Extensions (mingw-w64)'
 license=('PSF')
 arch=('any')
-mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang32' 'clang64' 'clangarm64')
```
[the remark this answers is not in the document]
Comment (lazka) [reply mingw-w64-python-pywin32/PKGBUILD:13]: +1 for removal

**Comment by jeremyd2019** [comment]
> 2. clang doesn't have libadsiid.a

Doesn't it?  Shows up in https://packages.msys2.org/package/mingw-w64-clang-x86_64-crt-git?repo=clang64

**Comment by raedrizqie** [comment]
maybe some missing defs then

**Comment by jeremyd2019** [comment]
yeah it seems to be missing several GUIDs.  Don't know how it works for gcc, they don't seem to be present there either.

**Comment by raedrizqie** [comment]
> yeah it seems to be missing several GUIDs. Don't know how it works for gcc, they don't seem to be present there either.

GCC also don't have adsi extension because of this, but now I got it fixed locally (not in repo though). Pre b301 don't have those UUID I think.

Can't we just add windmc.exe for Clang CI? It's not linked to GCC anyway right?

**Comment by raedrizqie** [comment]
that's a good idea :)

**Comment by jeremyd2019** [comment]
And for the missing GUIDs, I sent https://sourceforge.net/p/mingw-w64/mailman/message/37364401/ but nobody seems to have jumped on adding them ;)

**Comment by raedrizqie** [comment]
define those GUIDs in adsi.i will fix the problem :)

**Comment by raedrizqie** [comment]
maybe Kai could help

**Comment by jeremyd2019** [comment]
https://sourceforge.net/p/mingw-w64/mailman/message/37370098/

**Comment by raedrizqie** [comment]
Nice. But is that enough? We still need to define every constant for each missing GUID right?

**Comment by raedrizqie** [comment]
`(except for the 5 that were also missing from the Windows SDK, but since they're also missing from the Windows SDK hopefully nobody uses them 😉)`

this is what I meant by not enough.. we still have to define the constant of their GUID

**Comment by raedrizqie** [comment]
I think we still need to define something like this for example:

`const GUID CLSID_ADsDSOObject = {0x54, 0x93, 0x65, 0xD0, 0xEC, 0x26, 0x11, 0xCF, 0x83, 0x10, 0x00, 0xAA, 0x00, 0xB5, 0x05, 0xDB};`

the constant value of those GUIDs are missing from the headers.

**Comment by jeremyd2019** [comment]
They don't need to be in the headers, they will be in libadsiid.a.

**Comment by Biswa96** [comment]
But if those GUIDs are part of any COM interface, class or typelib those will go in a header. Then the header got included in .c file and compiled.

**Pull request merged**